### PR TITLE
mobile: Fix bugs in the Apple PAC proxy resolver

### PR DESCRIPTION
**Pull request by abeyad**

### CODE REVIEW

File: mobile/test/swift/integration/proxying/HTTPRequestUsingPacProxyTest.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      .respectSystemProxySettings(true)
+      .build()
+
+    let pacScript = """
+    function FindProxyForURL(url, host) {
+      return "PROXY 127.0.0.1:\(proxyPort)";
+    }
+    """;
+    EnvoyTestServer.setHeadersAndData("Content-Type", header_value: "application/x-ns-proxy-autoconfig", response_body: pacScript)
+    EnvoyTestApi.registerTestProxyResolver("127.0.0.1", port: httpPort, usePacResolver: true)
```
Comment (DavidSchinazi) [inline mobile/test/swift/integration/proxying/HTTPRequestUsingPacProxyTest.swift:79]: I don't understand where this test tells the stack to use a PAC file. I went looking through the source for `EnvoyTestApi` but didn't find it there either. For this to work you're going to have to tell the stack to use a certain PAC file, and then you need something to serve that PAC file

File: mobile/test/common/proxy/test_apple_proxy_settings_monitor.cc [hunk 2]
```diff
@@ -44,13 +46,11 @@ CFDictionaryRef TestAppleSystemProxySettingsMonitor::getSystemProxySettingsWithP
   const void* keys[] = {kCFNetworkProxiesProxyAutoConfigEnable,
                         kCFNetworkProxiesProxyAutoConfigURLString};
 
+  // Interpret the host + port as the location from which to obtain the PAC file.
+  const std::string pac_file_url = absl::StrCat("http://", host_, ":", port_);
```
Comment (DavidSchinazi) [inline mobile/test/common/proxy/test_apple_proxy_settings_monitor.cc:50]: This isn't a valid URL, as it has no path. Doesn't it need the `.pac` file format?
```suggestion
  const std::string pac_file_url = absl::StrCat("http://", host_, ":", port_, "/proxy.pac");
```

File: mobile/test/swift/integration/proxying/HTTPRequestUsingPacProxyTest.swift (hunk 1 above)
Comment (abeyad) [reply mobile/test/swift/integration/proxying/HTTPRequestUsingPacProxyTest.swift:79]: This part ensures that the `proxy_autoconfiguration_file_url` here: https://github.com/envoyproxy/envoy/pull/36477/files#diff-3c1c391c9cebec316f969380f3858fccec0afbcac44ffdc04c9a3097a24e1ee7R97 is set to `127.0.0.1:{httpPort}`.  I've printed the `cf_proxy_autoconfiguration_file_url` here https://github.com/envoyproxy/envoy/pull/36477/files#diff-3c1c391c9cebec316f969380f3858fccec0afbcac44ffdc04c9a3097a24e1ee7R89 that is passed to `CFNetworkExecuteProxyAutoConfigurationURL` and the value is indeed `127.0.0.1:{httpPort}`.  So the `CFNetworkExecuteProxyAutoConfigurationURL` API should be trying to fetch the PAC script from `127.0.0.1:{httpPort}`.

File: mobile/test/common/proxy/test_apple_proxy_settings_monitor.cc (hunk 2 above)
Comment (abeyad) [reply mobile/test/common/proxy/test_apple_proxy_settings_monitor.cc:50]: In this test, the HTTP server at `127.0.0.1:{httpPort}` is configured to return the PAC script in the response body (https://github.com/envoyproxy/envoy/pull/36477/files#diff-e49e7f9a928756a0062b93c2e3d990ca0e40f2b89f4d31f29693a2bc2a861b9dR78), with the `Content-Type` set to `application/x-ns-proxy-autoconfig`.  The reason I thought this should work is we do the same thing for our Android PAC resolver test: https://github.com/envoyproxy/envoy/blob/bfa0e07aab8500f49eb54766be7ae4edd1d15de3/mobile/test/kotlin/integration/proxying/ProxyPollPerformHTTPRequestWithoutUsingPACProxyTest.kt#L52-L63

Comment (abeyad) [reply mobile/test/common/proxy/test_apple_proxy_settings_monitor.cc:50]: I tried adding `proxy.pac` to the path in the URL, and I still get the same issue.

**Pull request closed without merging**